### PR TITLE
Kafka output producer, send telegraf metrics to Kafka broker(s)

### LINKED ISSUE #38: Support sending metrics to Kafka
We should be able to send metrics to a variety of places other than InfluxDB. This means that we'll need to have some sort of framework for defining new output sinks. This also means that we should be able to disable sending metrics to any of these output sinks, including InfluxDB.

What I'm thinking is to pull the InfluxDB settings into another area like `outputs`. Then, like plugins, define a number of different outputs where metrics can be sent to.

For Kafka, we should send metrics using the line protocol. This should be fairly simply using the InfluxDB client to convert the metrics to their line protocol equivalents.

The work to add support for Riemann (#34) will also need this output sink implementation.

### PR DESCRIPTION
**Pull request by sparrc**
Closes #38

/cc @pauldix

### CODE REVIEW

File: agent.go [hunk 1]
```diff
@@ -233,17 +242,17 @@ func (a *Agent) crankSeparate(shutdown chan struct{}, plugin *runningPlugin) err
 
 		bp.Prefix = plugin.name + "_"
 		bp.Config = plugin.config
-		err := plugin.plugin.Gather(&bp)
-		if err != nil {
-			return err
+		gathererr := plugin.plugin.Gather(&bp)
```
Comment (pauldix) [inline agent.go]: Should still use `err` here and in the flush code below. Use this pattern instead:

```go
if err := plugin.plugin.Gather(&bp); err != nil {
  log.Printf("Error in plugin [%s]: %s", plugin.name, err.Error())
}
```

File: agent.go [hunk 2]
```diff
@@ -258,16 +267,21 @@ func (a *Agent) crankSeparate(shutdown chan struct{}, plugin *runningPlugin) err
 func (a *Agent) flush(bp *BatchPoints) error {
 	var wg sync.WaitGroup
 	var outerr error
+
 	for _, o := range a.outputs {
 		wg.Add(1)
 		go func(ro *runningOutput) {
 			defer wg.Done()
-			outerr = ro.output.Write(bp.BatchPoints)
+			err := ro.output.Write(bp.BatchPoints)
```
Comment (pauldix) [inline agent.go]: same here, use the `if err := ...; err != nil` pattern

File: agent.go [hunk 3]
```diff
@@ -351,7 +365,7 @@ func (a *Agent) Run(shutdown chan struct{}) error {
 	for {
 		err := a.crankParallel()
 		if err != nil {
-			log.Printf("Error in plugins: %s", err)
+			log.Printf(err.Error())
```
Comment (pauldix) [inline agent.go]: same err pattern here

**Comment by pauldix** [comment]
Just the few comments to address, otherwise looks good.

**Comment by sparrc** [comment]
great, making those changes now, thanks @pauldix

**Pull request merged**